nixos/modules/services/hardware/evdevremapkeys.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.evdevremapkeys;
+
+  fullConfig = {
+    devices = cfg.devices;
+  };
+
+  cfgFile = pkgs.writeText "config.yaml" (builtins.toJSON fullConfig);
+in {
+
+  options.services.evdevremapkeys = {
+
+    # Note: this corresponds to the YAML configuration that evdevremap keys
+    # expects.
+    devices = mkOption {
+      default = {};
+      description = "Input devices to remap using evdevremapkeys";
+      type = with types; listOf (submodule {
+        options = {
+
+          input_name = mkOption {
+            type = str;
+            description = ''
+              Name of device to be remapped (from `xinput list`).
+            '';
+          };
+
+          output_name = mkOption {
+            type = str;
+            description = ''
+              Name of virtual device that will output remapped events.
+            '';
+          };
+
+          remappings = mkOption {
+            type = attrsOf (listOf str);
+            description = ''
+              Attrset of inputted event name to list of events that will be
+              fired instead. The event names are standard Linux input event
+              names, for a list see include/uapi/linux/input-event-codes.h in
+              the Linux source tree.
+
+              For example:
+                remappings = {
+                  KEY_ESC = [
+                    "KEY_A"
+                    "KEY_B"
+                  ];
+                };
+              will remap the escape key to instead fire an A and B keypress
+              simultaneously when pressed.
+            '';
+          };
+        };
+      });
+    };
+
+  };
+
+  config = mkIf (cfg.devices != []) {
+
+    systemd.services.evdevremapkeys = {
+      description = "Remap Evdev Input";
+      after = [ "systemd-udev-settle.service" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
```

Comment:
It would be nice to add some `systemd` hardening options here.

Reply:
What do you mean?

Reply:
I think he means:

Can we run the daemon as a `DynamicUser=yes`  instead of as `root`

Also we could set things like:

```
RestrictAddressFamilies=AF_NETLINK
PrivateTmp=yes
```

Etc.. (See `man systemd.exec` for more isolation options)

However I expect that this unit probably _has_ to run as root to be able to create new input events

Reply:
I'm going to take a look if this is possible to harden by running as a less privileged user. But I'm also not expecting much.

Reply:
Even if you still end up running as `root` there are some settings you can apply to limit the abilities this service can have on a system. Though still hoping for a non `root` user, maybe with some `capabilities` or something.